geomesa-utils-parent/geomesa-utils/src/test/scala/org/locationtech/geomesa/utils/io/fs/LocalFileHandleTest.scala

```diff
@@ -0,0 +1,70 @@
+/***********************************************************************
+ * Copyright (c) 2013-2024 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.utils.io.fs
+
+import org.apache.commons.io.IOUtils
+import org.junit.runner.RunWith
+import org.locationtech.geomesa.utils.collection.SelfClosingIterator
+import org.locationtech.geomesa.utils.io.WithClose
+import org.locationtech.geomesa.utils.io.fs.LocalDelegate.{CachingStdInHandle, StdInHandle}
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+import java.io.{BufferedInputStream, ByteArrayInputStream, IOException, InputStream}
+
+@RunWith(classOf[JUnitRunner])
+class LocalFileHandleTest extends Specification {
+
+  "LocalFileHandle" should {
+    "cache bytes when reading from stdin" in {
+      @throws[IOException]
+      def readNBytes(stream: InputStream, n: Int): Array[Byte] = {
+        val buffer = new Array[Byte](n)
+        var totalBytesRead = 0
+
+        while (totalBytesRead < n) {
+          val bytesRead = stream.read(buffer, totalBytesRead, n - totalBytesRead)
+          if (bytesRead == -1) {
+            throw new IOException("End of stream reached before reading 'n' bytes.")
```

Comment:
It might be useful to print out the value of `n`:

```scala
throw new IOException(s"End of stream reached before reading ${n} bytes.")
```

Reply:
https://github.com/locationtech/geomesa/pull/3036/commits/10b6ed596f780781077cad0ff9c47210886abc84